src/read_write/ply.rs

```diff
@@ -315,83 +331,134 @@ impl PlyIterator {
-                        prop.data_type,
-                        |p: &mut Point, val: u8| p.color.red = val,
+                        prop,
+                        |p: &mut PointsBatch, _: &str, val: u8| {
+                            let vec: &mut Vec<Vector3<u8>> =
+                                p.get_attribute_vec_mut("color").unwrap();
+                            vec.last_mut().unwrap().x = val
+                        },
                         &mut num_bytes_per_point
                     ));
                 }
                 "g" | "green" => {
                     readers.push(read_casted_property!(
-                        prop.data_type,
-                        |p: &mut Point, val: u8| p.color.green = val,
+                        prop,
+                        |p: &mut PointsBatch, _: &str, val: u8| {
+                            let vec: &mut Vec<Vector3<u8>> =
+                                p.get_attribute_vec_mut("color").unwrap();
+                            vec.last_mut().unwrap().y = val
+                        },
                         &mut num_bytes_per_point
                     ));
                 }
                 "b" | "blue" => {
                     readers.push(read_casted_property!(
-                        prop.data_type,
-                        |p: &mut Point, val: u8| p.color.blue = val,
+                        prop,
+                        |p: &mut PointsBatch, _: &str, val: u8| {
+                            let vec: &mut Vec<Vector3<u8>> =
+                                p.get_attribute_vec_mut("color").unwrap();
+                            vec.last_mut().unwrap().z = val
+                        },
                         &mut num_bytes_per_point
                     ));
                 }
                 "a" | "alpha" => {
-                    readers.push(read_casted_property!(
-                        prop.data_type,
-                        |p: &mut Point, val: u8| p.color.alpha = val,
-                        &mut num_bytes_per_point
-                    ));
-                }
-                "intensity" => {
-                    readers.push(read_casted_property!(
-                        prop.data_type,
-                        |p: &mut Point, val| p.intensity = Some(val),
-                        &mut num_bytes_per_point
-                    ));
+                    readers.push(create_skip_fn!(prop, &mut num_bytes_per_point, 1));
                 }
-                other => {
-                    println!("Will ignore property '{}' on 'vertex'.", other);
+                _ => {
                     use self::DataType::*;
                     match prop.data_type {
-                        Uint8 | Int8 => readers.push(create_skip_fn!(&mut num_bytes_per_point, 1)),
-                        Uint16 | Int16 => {
-                            readers.push(create_skip_fn!(&mut num_bytes_per_point, 2))
+                        Uint8 => {
```

Comment:
does the ply now also support more than color, position and intensity? Great!

Reply:
Yes, it reads everything we can put into our `AttributeData`. I didn't implement multidimensional attributes other than color and position though.